### DISCUSSION TITLE
Confusion regarding intended behavior with log level

### DISCUSSION THREAD

**Discussion by hunnorth** [aws/aws-sdk-cpp#2647] [Q&A]
I have been using this configuration in my `SDKOptions` to configure my logging (pattern seen [here](https://github.com/awslabs/aws-lambda-cpp/blob/master/examples/dynamodb/main.cpp#L211-L212)):
```
Aws::SDKOptions options;
options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Info;
options.loggingOptions.logger_create_fn = []() {
  return Aws::MakeShared<Aws::Utils::Logging::ConsoleLogSystem>(
    "MyLogSystem", Aws::Utils::Logging::LogLevel::Info);
};
// Prevent CRT from getting INFO log level from above
// See: https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/logging.html
options.loggingOptions.crt_logger_create_fn = []() {
  return Aws::MakeShared<Aws::Utils::Logging::DefaultCRTLogSystem>(
    "CRTLogSystem", Aws::Utils::Logging::LogLevel::Warn);
};
```
Unfortunately, my logs were showing the following unexpected behavior:
- Regular logs were not obeying the Info level debugging (showing debug)
- CRT logs were not obeying the configured Warn level debugging (showing info and lower)
- The log level setting seemed to have no effect. As mentioned in previous bullet points, seeing debug log level, etc.

I am however able to achieve the desired behavior by removing this line:
`options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Info;`

Why is this the case? Observing `src/aws-cpp-sdk-core/include/aws/core/Aws.h`, the `LoggingOptions` struct specifies that that `logger_create_fn` take precedence over the `logLevel` field:
```
/**
 * Defaults to empty, if logLevel has been set and this field is empty, then the default log system will be used.
 * otherwise, we will call this closure to create a logger
 */
std::function<std::shared_ptr<Aws::Utils::Logging::LogSystemInterface>()> logger_create_fn;
```
Maybe I am misinterpreting something, but it seems to me that using both `logLevel` and the `logger_create_fn` breaks the intended behavior of logging at a specific level.

Thank you

**Comment by jmklix** [comment]
I think you are understanding it correctly, I'm just not getting the same logs that you seem to be getting. Can you paste a snippet of the DEBUG logs that you are getting when you set the loglevel to INFO? what version of this sdk are you using? and a minimal sample that reproduces this would be helpful
